pkg/receive/handler.go

```diff
@@ -863,6 +869,33 @@ func (h *Handler) fanoutForward(ctx context.Context, params remoteWriteParams) (
 	}
 }
 
+func printMap(data map[endpointReplica]map[string]trackedSeries) {
+	for key, innerMap := range data {
+		fmt.Printf("Endpoint: %s, CapNProtoAddress: %s, AZ: %s, Replica: %d\n", key.endpoint.Address, key.endpoint.CapNProtoAddress, key.endpoint.AZ, key.replica)
+		for strKey, series := range innerMap {
+			fmt.Printf("  Key: %s\n", strKey)
+			fmt.Printf("    SeriesIDs: %v\n", series.seriesIDs)
+			fmt.Printf("    TimeSeries length: %d\n", len(series.timeSeries))
+		}
+	}
+}
+
+func (h *Handler) distributeTimeseriesToReplicasIngestorOnly(tenantHTTP string, replicas []uint64, timeseries []prompb.TimeSeries) (map[endpointReplica]map[string]trackedSeries, map[endpointReplica]map[string]trackedSeries, error) {
+	remoteWrites := make(map[endpointReplica]map[string]trackedSeries)
```

Comment:
actually should avoid allocation if remote writes are empty, you can consider to reuse a static map for ingestor